sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/externalreport/TslintReportSensor.java

```diff
@@ -0,0 +1,232 @@
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.rules.RuleType;
+import org.sonar.api.utils.Version;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugin.typescript.TypeScriptLanguage;
+import org.sonar.plugin.typescript.TypeScriptRules;
+
+import static org.sonar.plugin.typescript.TypeScriptPlugin.TSLINT_REPORT_PATHS;
+
+public class TslintReportSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(TslintReportSensor.class);
+  private final Gson gson = new Gson();
+
+  private final Set<String> activatedTslintKeys = new HashSet<>();
+
+  private static final Set<String> BUG_TSLINT_RULES = new HashSet<>(Arrays.asList(
+    "await-promise",
+    "ban-comma-operator",
+    "ban",
+    "curly",
+    "forin",
+    "import-blacklist",
+    "label-position",
+    "no-arg",
+    "no-bitwise",
+    "no-conditional-assignment",
+    "no-console",
+    "no-construct",
+    "no-debugger",
+    "no-duplicate-super",
+    "no-duplicate-switch-case",
+    "no-duplicate-variable",
+    "no-dynamic-delete",
+    "no-empty",
+    "no-eval",
+    "no-floating-promises",
+    "no-for-in-array",
+    "no-implicit-dependencies",
+    "no-inferred-empty-object-type",
+    "no-invalid-template-strings",
+    "no-invalid-this",
+    "no-misused-new",
+    "no-null-keyword",
+    "no-object-literal-type-assertion",
+    "no-return-await",
+    "no-shadowed-variable",
+    "no-sparse-arrays",
+    "no-string-literal",
+    "no-string-throw",
+    "no-submodule-imports",
+    "no-switch-case-fall-through",
+    "no-this-assignment",
+    "no-unbound-method",
+    "no-unnecessary-class",
+    "no-unsafe-any",
+    "no-unsafe-finally",
+    "no-unused-expression",
+    "no-unused-variable",
+    "no-use-before-declare",
+    "no-var-keyword",
+    "no-void-expression",
+    "prefer-conditional-expression",
+    "prefer-object-spread",
+    "radix",
+    "restrict-plus-operands",
+    "strict-boolean-expressions",
+    "strict-type-predicates",
+    "switch-default",
+    "triple-equals",
+    "typeof-compare",
+    "use-default-type-parameter",
+    "use-isnan"));
+
+  public TslintReportSensor(CheckFactory checkFactory) {
+    TypeScriptRules typeScriptRules = new TypeScriptRules(checkFactory);
+    typeScriptRules.forEach(typeScriptRule -> {
+      if (typeScriptRule.isEnabled()) {
+        activatedTslintKeys.add(typeScriptRule.tsLintKey());
+      }
+    });
+  }
+
+  @Override
+  public void execute(SensorContext context) {
+    boolean externalIssuesSupported = context.getSonarQubeVersion().isGreaterThanOrEqual(Version.create(7, 2));
+    String[] tslintReportPaths = context.config().getStringArray(TSLINT_REPORT_PATHS);
+
+    if (tslintReportPaths.length == 0) {
+      return;
+    } else if (!externalIssuesSupported) {
+      LOG.error("Current version of SonarQube doesn't support import of external issues (at least 7.2 required).");
```

Comment:
"Current version" is bit ambiguous here. I would rephrase as "Import of external issues requires SonarQube 7.2 or greater."